source/expressions.tex

```diff
@@ -5540,10 +5540,13 @@
 \pnum
 If the keyword \keyword{delete} in a \grammarterm{delete-expression}
 is not preceded by the unary \tcode{::} operator and the type of the operand is
-a pointer to a (possibly cv-qualified) class type \tcode{T}:
+a pointer to a (possibly cv-qualified) class type \tcode{T}
+or (possibly multidimensional) array thereof:
 \begin{itemize}
 \item
-If \tcode{T} has a virtual destructor,
+For a single-object delete expression,
+if the operand is a pointer to \cv{} \tcode{T} and
```

Comment:
I still want to refer this example, but a bit difference
````cpp
auto ptr = new A[2];
ptr++; // now, ptr point to the `1` element of the array, and its an object of type `A`
delete ptr; 
````
It seems `delete ptr; ` will cause UB in most implementations. The operand of the single-object delete expression should be restricted to that produced by single-object new-expression, however it didn't represent in [expr.delete#2].

Reply:
I agree this should say "complete object". That's not this issue, though.